pkg/handler/v2/external_cluster/eks.go

```diff
@@ -134,28 +152,208 @@ func getEKSNodePools(ctx context.Context, cluster *kubermaticapiv1.ExternalClust
+	accessKeyID, secretAccessKey, err := awsprovider.GetCredentialsForEKSCluster(*cloudSpec, secretKeySelector)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := awsprovider.GetClientSet(accessKeyID, secretAccessKey, cloudSpec.EKS.Region)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterName := cloudSpec.EKS.Name
+
+	nodeGroupInput := &eks.DescribeNodegroupInput{
+		ClusterName:   &clusterName,
+		NodegroupName: &nodeGroupName,
+	}
+
+	nodeGroupOutput, err := client.EKS.DescribeNodegroup(nodeGroupInput)
+	if err != nil {
+		return nil, err
+	}
+	nodeGroup := nodeGroupOutput.Nodegroup
+
+	nodes, err := clusterProvider.ListNodes(cluster)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	var readyReplicas int32
+	for _, n := range nodes.Items {
+		if n.Labels != nil {
+			if n.Labels[GKENodepoolNameLabel] == nodeGroupName {
+				readyReplicas++
+			}
+		}
+	}
+	machineDeployment := nodeGroupToMD(nodeGroup, readyReplicas)
+
+	return &machineDeployment, err
+}
+
+func nodeGroupToMD(nodeGroup *eks.Nodegroup, readyReplicas int32) apiv2.ExternalClusterMachineDeployment {
+	return apiv2.ExternalClusterMachineDeployment{
+		NodeDeployment: apiv1.NodeDeployment{
+			ObjectMeta: apiv1.ObjectMeta{
+				ID:   *nodeGroup.NodegroupName,
+				Name: *nodeGroup.NodegroupName,
+			},
+			Spec: apiv1.NodeDeploymentSpec{
+				Replicas: int32(*nodeGroup.ScalingConfig.DesiredSize),
+				Template: apiv1.NodeSpec{
+					Versions: apiv1.NodeVersionInfo{
+						Kubelet: *nodeGroup.Version,
 					},
 				},
-				Status: clusterv1alpha1.MachineDeploymentStatus{
-					Replicas:      int32(*nodeGroup.ScalingConfig.DesiredSize),
-					ReadyReplicas: readyReplicas,
-				},
 			},
+			Status: clusterv1alpha1.MachineDeploymentStatus{
+				Replicas:      int32(*nodeGroup.ScalingConfig.DesiredSize),
+				ReadyReplicas: readyReplicas,
+			},
+		},
+	}
+}
+
+func patchEKSMD(ctx context.Context, old, new *apiv2.ExternalClusterMachineDeployment, secretKeySelector provider.SecretKeySelectorValueFunc, cloudSpec *kubermaticapiv1.ExternalClusterCloudSpec) (*apiv2.Update, error) {
+
+	accessKeyID, secretAccessKey, err := awsprovider.GetCredentialsForEKSCluster(*cloudSpec, secretKeySelector)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := awsprovider.GetClientSet(accessKeyID, secretAccessKey, cloudSpec.EKS.Region)
+	if err != nil {
+		return nil, err
+	}
+	clusterName := cloudSpec.EKS.Name
+	nodeGroupName := new.NodeDeployment.Name
+
+	desiredReplicas := new.NodeDeployment.Spec.Replicas
+	updateOutput, err := resizeEKSNodeGroup(ctx, client, clusterName, nodeGroupName, int64(desiredReplicas))
+	if err != nil {
+		return nil, err
+	}
+	return updateOutput, err
```

Comment:
should return `new apiv2.ExternalClusterMachineDeployment` instead of apiv2.Update
Please get rid of `apiv2.Update`